client/packages/invoices/src/Returns/modals/InboundReturn/useDraftInboundReturnLines.ts

```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from 'react';
+import { InboundReturnLine, RecordPatch } from '@openmsupply-client/common';
+import { useReturns } from '../../api';
+
+export const useDraftInboundReturnLines = (stockLineIds: string[]) => {
+  const [draftLines, setDraftLines] = React.useState<InboundReturnLine[]>([]);
+
+  const lines = useReturns.lines.inboundReturnLines(stockLineIds);
+
+  useEffect(() => {
+    const newDraftLines = (lines ?? []).map(seed => ({ ...seed }));
+
+    setDraftLines(newDraftLines);
+  }, [lines]);
+
+  const update = (patch: RecordPatch<InboundReturnLine>) => {
+    setDraftLines(currLines => {
+      const newLines = currLines.map(line => {
+        if (line.id !== patch.id) {
+          return line;
+        }
+        return { ...line, ...patch, isUpdated: true };
```

Comment:
We don't really need to keep track of isUpdated ? It's a bit hard to really trully figure out if it's been udpated or not (have to constantly compary to full previous state)